sentry-android-core/src/main/java/io/sentry/android/core/util/ConnectivityChecker.java

```diff
@@ -16,98 +17,163 @@
   }
 
+  /**
+   * Return the Connection status
+   *
+   * @param context the Context
+   * @param connectivityManager the ConnectivityManager
+   * @param logger the Logger
+   * @return true if connected or no permission to check, false otherwise
+   */
   @SuppressWarnings({"deprecation", "MissingPermission"})
-  private static Boolean isConnected(
-      Context context, ConnectivityManager connectivityManager, ILogger logger) {
-    android.net.NetworkInfo activeNetwork =
-        getActiveNetworkInfo(context, connectivityManager, logger);
-
-    if (activeNetwork == null) {
-      logger.log(SentryLevel.INFO, "NetworkInfo is null and cannot check network status");
-      return null;
-    }
-    return activeNetwork.isConnected();
-  }
-
-  @SuppressWarnings({"deprecation", "MissingPermission"})
-  private static @Nullable android.net.NetworkInfo getActiveNetworkInfo(
-      @NotNull Context context,
-      @NotNull ConnectivityManager connectivityManager,
-      @NotNull ILogger logger) {
+  private static @NotNull ConnectivityChecker.Status getConnectionStatus(
+      final @NotNull Context context,
+      final @NotNull ConnectivityManager connectivityManager,
+      final @NotNull ILogger logger) {
     if (!Permissions.hasPermission(context, Manifest.permission.ACCESS_NETWORK_STATE)) {
       logger.log(SentryLevel.INFO, "No permission (ACCESS_NETWORK_STATE) to check network status.");
-      return null;
+      return Status.NO_PERMISSION;
     }
+    final android.net.NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
 
-    return connectivityManager.getActiveNetworkInfo();
+    if (activeNetworkInfo == null) {
+      logger.log(SentryLevel.INFO, "NetworkInfo is null, there's no active network.");
+      return Status.NOT_CONNECTED;
+    }
+    return activeNetworkInfo.isConnected() ? Status.CONNECTED : Status.NOT_CONNECTED;
   }
 
   /**
    * Check the connection type of the active network
    *
    * @param context the App. context
    * @param logger the logger from options
+   * @param buildInfoProvider the BuildInfoProvider provider
    * @return the connection type wifi, ethernet, cellular or null
    */
-  @SuppressLint({"ObsoleteSdkInt", "MissingPermission"})
-  public static String getConnectionType(@NotNull Context context, @NotNull ILogger logger) {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
-      if (connectivityManager == null) {
-        return null;
-      }
-      if (!Permissions.hasPermission(context, Manifest.permission.ACCESS_NETWORK_STATE)) {
-        logger.log(
-            SentryLevel.INFO, "No permission (ACCESS_NETWORK_STATE) to check network status.");
-        return null;
-      }
-      Network activeNetwork = connectivityManager.getActiveNetwork();
+  @SuppressLint({"ObsoleteSdkInt", "MissingPermission", "NewApi"})
+  public static @Nullable String getConnectionStatus(
```

Comment:
`m`: Shouldn't this still be `getConnectionStatus`?

Reply:
yeah the IDE renamed this one too, damn, good catch